python/ray/data/_internal/execution/backpressure_policy.py

```diff
@@ -113,3 +134,70 @@ def can_run(self, op: "PhysicalOperator") -> bool:
                 f"Concurrency cap for {op} increased to {self._concurrency_caps[op]}"
             )
         return metrics.num_tasks_running < self._concurrency_caps[op]
+
+
+class StreamingOutputBackpressurePolicy(BackpressurePolicy):
+    """A backpressure policy that throttles the streaming outputs of the `DataOpTask`s.
+
+    The are 2 levels of configs to control the behavior:
+    - At the Ray Core level, we use
+      `MAX_BLOCKS_IN_GENERATOR_BUFFER` to limit the number of blocks buffered in
+      the streaming generator of each OpDataTask. When it's reached, the task will
+      be blocked at `yield`.
+    - At the Ray Data level, we use
+      `MAX_BLOCKS_IN_GENERATOR_BUFFER` to limit the number of blocks buffered in the
+      output queue of each operator. When it's reached, we'll stop reading from the
+      streaming generators of the op's tasks, and thus trigger backpressure at the
+      Ray Core level.
```

Comment:
Can you also add a note here what the total buffered output size per op is expected to be?